settings.gradle

```diff
@@ -1,13 +1,20 @@
 rootProject.name = 'paparazzi-root'
 
+include ':libs:layoutlib'
+include ':libs:native-macarm'
+include ':libs:native-macosx'
+include ':libs:native-win'
+include ':libs:native-linux'
+include ':paparazzi'
+include ':paparazzi-agent'
+include ':paparazzi-gradle-plugin'
+
 include ':sample'
 
-includeBuild('paparazzi') {
+enableFeaturePreview('TYPESAFE_PROJECT_ACCESSORS')
+
+includeBuild('build-logic') {
```

Comment:
Note: detekt also has a build that runs on itself. When you do `gradlew detekt` in the detekt repo, it builds the plugin, the rules and then runs those just-built rules on the same rules.

There, the plugin is included directly [along with build-logic](https://github.com/detekt/detekt/blob/main/settings.gradle.kts#L5-L6), not transitively, so that build-logic can be also used inside the plugin.